### PR TITLE
Added every_before_after event filter

### PR DESCRIPTION
**Pull request by theory-in-progress**
Related to #2853 

Description:

* New Feature
* Executes after every `every` Iterations
* For after=8, before=25, every=5 Executes on Iterations 9 14 19 24

Check list:

- [x] New tests are added (if a new feature is added)
- [x] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

**Review by vfdev-5 — COMMENTED** [review]
@theory-in-progress thanks for the PR ! Please add tests, I think args checking wont pass

File: ignite/engine/events.py [hunk 1]
```diff
@@ -71,14 +71,17 @@ def __call__(
...
+                (
+                    event_filter is not None,
+                    once is not None,
+                    (every is not None or before is not None or after is not None),
+                )
+            )
+            != 1
         ):
-            raise ValueError("All of the Arguments cannot be undefined")
+            raise ValueError("Only one of the Input Arguments should be Sepcified, except before, after and every")
```
Comment (vfdev-5) [inline ignite/engine/events.py:84]: @theory-in-progress please rewrite the error message and fix typos and capitalize only the word of the sentence.

File: tests/ignite/engine/test_custom_events.py [hunk 2]
```diff
@@ -129,23 +129,26 @@ def process_func(engine, batch):
...
+                        )
+                    except ValueError:
+                        with pytest.raises(
+                            ValueError,
+                            match=r"Only one of the input arguments should be specified, "
+                            "except before, after and every",
+                        ):
+                            Events.ITERATION_STARTED(
+                                event_filter=event_filter, once=once, every=every, before=before, after=after
+                            )
```
Comment (vfdev-5) [inline tests/ignite/engine/test_custom_events.py:151]:
```suggestion
    def ef(e, i):
        return 1

    expected_raise = {
        # event_filter, every, once, before, after
        (None, None, None, None, None): True,  # raises ValueError
        (ef, None, None, None, None): False,
        (None, 2, None, None, None): False,
        (ef, 2, None, None, None): True,

        (None, None, 2, None, None): False,
        (ef, None, 2, None, None): True,
        (None, 2, 2, None, None): True,
        (ef, 2, 2, None, None): True,

        (None, None, None, 30, None): False,
        (ef, None, None, 30, None): True,
        (None, 2, None, 30, None): False,
        (ef, 2, None, 30, None): True,

        (None, None, 2, 30, None): True,
        (ef, None, 2, 30, None): True,
        (None, 2, 2, 30, None): True,
        (ef, 2, 2, 30, None): True,

        # event_filter, every, once, before, after
        (None, None, None, None, 10): False,
        (ef, None, None, None, 10): True,
        (None, 2, None, None, 10): False,
        (ef, 2, None, None, 10): True,

        (None, None, 2, None, 10): True,
        (ef, None, 2, None, 10): True,
        (None, 2, 2, None, 10): True,
        (ef, 2, 2, None, 10): True,

        (None, None, None, 25, 8): False,
        (ef, None, None, 25, 8): True,
        (None, 2, None, 25, 8): False,
        (ef, 2, None, 25, 8): True,

        (None, None, 2, 25, 8): True,
        (ef, None, 2, 25, 8): True,
        (None, 2, 2, 25, 8): True,
        (ef, 2, 2, 25, 8): True,
    }


    for event_filter in [None, ef]:
        for every in [None, 2]:
            for once in [None, 2]:
                for before, after in [(None, None), (None, 10), (30, None), (25, 8)]:

                    if expected_raise[(event_filter, every, once, before, after)]:                    
                        with pytest.raises(
                            ValueError,
                            match=r"Only one of the input arguments should be specified, "
                            "except before, after and every",
                        ):
                            Events.ITERATION_STARTED(
                                event_filter=event_filter, once=once, every=every, before=before, after=after
                            )
                    else:
                        Events.ITERATION_STARTED(
                            event_filter=event_filter, once=once, every=every, before=before, after=after
                        )
```

**Review by vfdev-5 — COMMENTED** [review]
Thanks for the update @theory-in-progress 
I left few comments

File: tests/ignite/engine/test_custom_events.py (hunk 2 above)
Comment (theory-in-progress) [reply tests/ignite/engine/test_custom_events.py:151]: It is better to predefine all the possible variations and then raise an error?

Comment (vfdev-5) [reply tests/ignite/engine/test_custom_events.py:151]: yes, this is my suggestion

File: ignite/engine/events.py [hunk 3]
```diff
@@ -301,6 +318,10 @@ def call_once(engine):
         def call_before(engine):
             # do something in 11 to 29 epoch
 
+        # e) once "every" iterations "before" and "after" evnt filter
```
Comment (vfdev-5) [inline ignite/engine/events.py:321]:
```suggestion
        # e) Mixing "every" and "before" / "after" event filters
```

File: ignite/engine/events.py [hunk 4]
```diff
@@ -301,6 +318,10 @@ def call_once(engine):
...
+        @engine.on(Events.EPOCH_STARTED(every=5, before=25, after=8))
+        def call_every_itr_before_after(engine):
+            # do something on 9, 14, 19, 24 iterations
```
Comment (vfdev-5) [inline ignite/engine/events.py:324]:
```suggestion
        # e) once "every" iterations "before" and "after" evnt filter
        @engine.on(Events.EPOCH_STARTED(every=5, before=25, after=8))
        def call_every_itr_before_after(engine):
            # do something on 9, 14, 19, 24 epochs
```

**Review by vfdev-5 — APPROVED** [review]
Thanks for the PR @theory-in-progress ! LGTM

**Pull request merged**